ps/ps.go

```diff
@@ -647,16 +777,60 @@ func processSECCOMP(p *process) (string, error) {
 
 // processLABEL returns the process label of process p.
 func processLABEL(p *process) (string, error) {
-	data, err := ioutil.ReadFile(fmt.Sprintf("/proc/%d/attr/current", p.pid))
+	data, err := ioutil.ReadFile(fmt.Sprintf("/proc/%s/attr/current", p.pid))
 	if err != nil {
 		if os.IsNotExist(err) {
 			// make sure the pid does not exist,
 			// could be system does not support labeling.
-			if _, err2 := os.Stat(fmt.Sprintf("/proc/%d", p.pid)); err2 != nil {
+			if _, err2 := os.Stat(fmt.Sprintf("/proc/%s", p.pid)); err2 != nil {
 				return "", errNoSuchPID
 			}
 		}
 		return "", err
 	}
 	return strings.Trim(string(data), "\x00"), nil
 }
+
+// findHostProcess returns the corresponding process from `hostProcesses` for
+// provided pid or nil if non is found.
+func findHostProcess(pid string) *process {
+	for _, hp := range hostProcesses {
+		// if it's a container, the 2nd NSpid points to the
+		// corresponding PID in the container's namespace
+		if len(hp.pstatus.nSpid) < 2 {
+			continue
+		}
+		if pid == hp.pstatus.nSpid[1] {
```

Comment:
is this check enough?  What happens if there are two containers and there are two different processes that have the same value in the second column but they are on different PID namespaces?

Reply:
So this would be two different pids running in the same cgroup but in different pid namespaces?

Reply:
In this case, the code is wrong. I think that using the creation of from `/proc/pid/stat` can be more useful then. To the best of my knowledge that's unique across the system as timers aren't multiplexed. WDYT?

Reply:
@rhatdan, something like:

```console
$ unshare -rpf sleep 1000 & unshare -rpf sleep 1000 &
$ $ pgrep sleep
24122
24124
$ grep NSpid /proc/24122/status
NSpid:	24122	1
$ grep NSpid /proc/24124/status
NSpid:	24124	1
```
the two sleep processes are running in two different PID namespaces, but they still show the same `NSpid` entry from the host

Reply:
@vrothberg I think we could just add another condition and check that the PID namespace is the same, if seen from the host or from the container PoV:

For example:
```console
$ sudo nsenter -m -t 23057 readlink -f /proc/1/ns/pid
/proc/1/ns/pid:[4026532540]

$ sudo readlink -f /proc/23057/ns/pid
/proc/23057/ns/pid:[4026532540]
```

The ID in the `[]` is the same in both cases.  That plus the code you already have should be enough IMO